extra_data/cli/make_virtual_cxi.py

```diff
@@ -4,24 +4,40 @@
 import os.path as osp
 import re
 import sys
+from textwrap import dedent
 
 from extra_data import RunDirectory
-from extra_data.components import AGIPD1M, LPD1M
+from extra_data.components import MPxDetectorBase
 from extra_data.exceptions import SourceNameError
 
 log = logging.getLogger(__name__)
 
 
 def _get_detector(data, min_modules):
-    for cls in (AGIPD1M, LPD1M):
+    for cls in MPxDetectorBase.__subclasses__():
         try:
             return cls(data, min_modules=min_modules)
         except SourceNameError:
             continue
 
 
+def _detectors():
+    """returns a list of names for all detector components available
+    """
+    return [d.__name__ for d in MPxDetectorBase.__subclasses__()]
+
+
 def main(argv=None):
-    ap = argparse.ArgumentParser('karabo-data-make-virtual-cxi')
+    example = dedent("""
+        Example:
+
+          extra-data-make-virtual-cxi -o ./out_file.h5 --min-modules 15
+          --fill-value data 0 --fill-value gain 1 /path/to/source/run
```

Comment:
I might end the first line with a `\` (escaped as `\\` in a Python string) and indent the second line a couple of spaces to make it obvious that it's one command split over two lines.